WalletWasabi.Fluent/ViewModels/Wallets/Send/TransactionPreviewViewModel.cs

```diff
@@ -22,75 +26,264 @@ public partial class TransactionPreviewViewModel : RoutableViewModel
-
-			var destinationAmount = transaction.CalculateDestinationAmount().ToDecimal(MoneyUnit.BTC);
-			var btcAmountText = $"{destinationAmount} bitcoins ";
-			var fiatAmountText = destinationAmount.GenerateFiatText(_wallet.Synchronizer.UsdExchangeRate, "USD");
-			AmountText = $"{btcAmountText}{fiatAmountText}";
 
 			AddressText = info.Address.ToString();
-
-			var fee = transaction.Fee;
-			var btcFeeText = $"{fee.ToDecimal(MoneyUnit.Satoshi)} sats ";
-			var fiatFeeText = fee.ToDecimal(MoneyUnit.BTC).GenerateFiatText(_wallet.Synchronizer.UsdExchangeRate, "USD");
-			FeeText = $"{btcFeeText}{fiatFeeText}";
-
 			PayJoinUrl = info.PayJoinClient?.PaymentUrl.AbsoluteUri;
 			IsPayJoin = PayJoinUrl is not null;
 
 			if (PreferPsbtWorkflow)
 			{
-				SkipCommand = ReactiveCommand.CreateFromTask(async () => await OnConfirmAsync(transaction));
+				SkipCommand = ReactiveCommand.CreateFromTask(async () => await OnConfirmAsync(_transaction));
+
 				NextCommand = ReactiveCommand.CreateFromTask(async () =>
 				{
-					var saved = await TransactionHelpers.ExportTransactionToBinaryAsync(transaction);
+					var saved = await TransactionHelpers.ExportTransactionToBinaryAsync(_transaction);
 
 					if (saved)
 					{
 						Navigate().To(new SuccessViewModel("The PSBT has been successfully created."));
 					}
 				});
+
 				_nextButtonText = "Save PSBT file";
 			}
 			else
 			{
-				NextCommand = ReactiveCommand.CreateFromTask(async () => await OnConfirmAsync(transaction));
+				NextCommand = ReactiveCommand.CreateFromTask(async () => await OnConfirmAsync(_transaction));
+
 				_nextButtonText = "Confirm";
 			}
 
+			AdjustFeeCommand = ReactiveCommand.CreateFromTask(async () =>
+			{
+				var feeRateDialogResult = await NavigateDialogAsync(new SendFeeViewModel(wallet, info, false));
+
+				if (feeRateDialogResult.Kind == DialogResultKind.Normal && feeRateDialogResult.Result != _info.FeeRate)
+				{
+					_info.FeeRate = feeRateDialogResult.Result;
+
+					var newTransaction = await BuildTransactionAsync();
+
+					if (newTransaction is { })
+					{
+						UpdateTransaction(newTransaction);
+					}
+				}
+			});
+
+			AvoidChangeCommand = ReactiveCommand.CreateFromTask(async () =>
+			{
+				var optimisePrivacyDialog =
+					await NavigateDialogAsync(new OptimisePrivacyViewModel(wallet, info, _transaction!));
+
+				if (optimisePrivacyDialog.Kind == DialogResultKind.Normal && optimisePrivacyDialog.Result is { })
+				{
+					UpdateTransaction(optimisePrivacyDialog.Result);
+				}
+			});
 		}
 
 		public bool PreferPsbtWorkflow => _wallet.KeyManager.PreferPsbtWorkflow;
 
-		public string AmountText { get; }
-
 		public string AddressText { get; }
 
-		public string FeeText { get; }
-
 		public string? PayJoinUrl { get; }
 
 		public bool IsPayJoin { get; }
 
+		public ICommand AdjustFeeCommand { get; }
+
+		public ICommand AvoidChangeCommand { get; }
+
+		private async Task<bool> InitialiseTransactionAsync()
+		{
+			var transactionInfo = _info;
```

Comment:
Does this make sense? 
`_info` is readonly so the reference is safe.

Reply:
done